httpx/_api.py

```diff
@@ -35,6 +35,7 @@ def request(
     verify: VerifyTypes = True,
     cert: CertTypes = None,
     trust_env: bool = True,
+    event_hooks: dict = None,
```

Comment:
I think I'd like us to prefer a stricter typing here, and a more straightforward style that requires using a list of callables:

```python
client = httpx.Client(event_hooks={"request": [on_request], "response": [on_response])
```

It would make it clearer that `client.event_hooks` is really just always a dict of lists, rather than "a dict of sometimes a callable, sometimes a list of callables, depending on what you've set".

Hence this suggestion (which would need to be applied across `_client.py` as well):

```suggestion
    event_hooks: Dict[str, List[Callable]] = None,
```